src/Table/Table.tsx

```diff
@@ -521,483 +617,540 @@ interface TableProps {
-    }, [props.searchProps, tableRowsImage]);
+        const doSearch = React.useCallback((): void => {
+            let searchResult: Array<TableRow> = [];
+            if (props.searchProps?.searchText && props.searchProps?.searchInColumns && props.searchProps?.searchInColumns?.length) {
+                searchResult = searchTextInArray(tableRowsImage, props.searchProps.searchText, props.searchProps.searchInColumns);
+            } else {
+                searchResult = [...tableRowsImage];
+            }
 
-    // useEffects ----------------------------------------------------------
+            setTableRows(searchResult);
+            props.searchProps?.onSearch && props.searchProps?.onSearch(searchResult);
+        }, [props.searchProps, tableRowsImage]);
 
-    // Adding event listener to listen to clicks outside the component on mount, removing on unmount
-    React.useEffect(() => {
-        document.addEventListener("mousedown", onClickOutside);
-        return () => {
-            document.removeEventListener("mousedown", onClickOutside);
-        };
-    });
+        // useEffects ----------------------------------------------------------
 
-    React.useEffect(() => {
-        if (!!props.onRowSelected) {
-            const notAllsAreRowsSelected: boolean = tableRows.some((row: TableRow) => !row.selected);
+        // Adding event listener to listen to clicks outside the component on mount, removing on unmount
+        React.useEffect(() => {
+            document.addEventListener("mousedown", onClickOutside);
+            return () => {
+                document.removeEventListener("mousedown", onClickOutside);
+            };
+        });
 
-            if (notAllsAreRowsSelected) {
-                setAllRowsChecked(false);
-            } else {
-                setAllRowsChecked(true);
+        React.useEffect(() => {
+            if (tableColumns?.length && tableRows?.length) {
+                const shouldFilter: boolean = tableColumns.some((column: TableHeader) => column.filters?.length);
+                if (shouldFilter) {
+                    const filteredRows: Array<TableRow> = filterArray(tableRowsImage, tableColumns);
+                    props.filterProps?.onAfterFilter && props.filterProps.onAfterFilter(filteredRows);
+                    setTableRows(filteredRows);
+                } else {
+                    setTableRows(tableRowsImage);
+                }
             }
-        }
+        }, [tableColumns]);
 
-        const actionColumnIsOpened: boolean = tableRows.some((row: TableRow) => {
-            return row.actionsDropdownDropped || row.subRows?.some((sub: TableRow) => sub.actionsDropdownDropped);
-        });
-        setAnOpenedAction(actionColumnIsOpened);
-    }, [currentTableRows]);
+        React.useEffect(() => {
+            if (!!props.onRowSelected) {
+                const notAllsAreRowsSelected: boolean = tableRows.some((row: TableRow) => !row.selected);
```

Comment:
`tableRows?.some` as well

Reply:
I have added some checks on the rows across the project to be  safer.